pkg/thing/interactors/register_thing.go

```diff
@@ -82,25 +82,27 @@ func (i *ThingInteractor) Register(authorization, id, name string) error {
 	if err != nil {
 		errReply := i.reply(id, "", err)
 		if errReply != nil {
-			i.logger.Error(errReply)
+			i.logger.Debug(err)
+			return errReply
 		}
-
 		return err
 	}
 
 	// Get the id generated as a token and send in the response
 	token, err := i.thingProxy.Create(id, name, authorization)
-	errReply := i.reply(id, token, err)
-	if errReply != nil {
-		i.logger.Error(errReply)
-		return errReply
+	if err != nil {
+		errReply := i.reply(id, token, err)
+		if errReply != nil {
+			i.logger.Debug(err)
+			return errReply
+		}
+		return err
 	}
 
-	err = i.connectorPublisher.SendRegisterDevice(id, name)
+	err = i.reply(id, token, err)
 	if err != nil {
 		i.logger.Error(err)
-		return err
 	}
 
-	return nil
+	return i.connectorPublisher.SendRegisterDevice(id, name)
```

Comment:
+1
Much better.

Reply:
I didn't do this because this change break the tests. However, I agree that it's better to understand and also will be according with `register` interactor.

Reply:
I've made some changes in patch, could you review it please? :+1: